surfsense_backend/app/services/document_processing/docling_service.py

```
@@ -0,0 +1,346 @@
+                    }
+                else:
+                    raise ValueError("No content could be extracted from document")
+            else:
+                raise ValueError("No document object returned by Docling")
+                
+        except Exception as e:
+            logger.error(f"❌ Docling processing failed for {filename}: {e}")
+            # Log the full error for debugging
+            import traceback
+            logger.error(f"Full traceback: {traceback.format_exc()}")
+            raise RuntimeError(f"Docling processing failed: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)